File: src/main/java/upbrella/be/util/CustomResponse.java
package upbrella.be.util;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class CustomResponse<T> {

    private String status;
    private int code;
    private String message;
    private T data;

    public CustomResponse(String status, int code, String message) {
        this.status = status;
        this.code = code;
        this.message = message;
    }
}

File: src/main/kotlin/upbrella/be/util/CustomResponse.kt
package upbrella.be.util

data class CustomResponse<T>(
    val status: String,
    val code: Int,
    val message: String,
    val data: T? = null
) {
    constructor(status: String, code: Int, message: String) : this(status, code, message, null)
}
